src/common/translate.ts

```diff
@@ -193,7 +192,7 @@ export class QuoteProcessor {
 
 const chineseLangCodes = ['zh-Hans', 'zh-Hant', 'lzh', 'yue', 'jdbhw', 'xdbhw']
 
-export async function translate(query: TranslateQuery) {
+export async function translate(query: TranslateQuery, engine: IEngine | undefined) {
```

Comment:
‌‌‌Why is this engine singled out?

Reply:
Because maintaining a long-term connection is necessary when using WebSockets, we don't have to re-register the WebSocket every time a request is made. To ensure the WebSocket's long-term existence, we need a long-term existing engine, rather than generating a new engine every time the translate function is used, as was done previously.

Reply:
‌‌‌‌Can this `engine` be used as a field in `TranslateQuery`?

Reply:
> ‌‌‌‌Can this `engine` be used as a field in `TranslateQuery`? It should be fine.  I'll give it a try tomorrow when I have time . Also, OpenAI has updated their request specifications, so we need to make some minor adjustments too.

Reply:
It should be fine. I'll give it a try tomorrow when I have time . Also, OpenAI has updated their request specifications, so we need to make some minor adjustments too.

Reply:
I've noticed that setting the engine used as a field in TranslateQuery isn't feasible. This is because the engine needs to exist as a useState to avoid being rebuilt every time the translate method is called. Unless you plan to revert to the previous WebAPI Class approach, it's preferable to continue passing the engine as the second parameter, as we're currently doing.